code-editor/scripts/README.md

```diff
@@ -0,0 +1,74 @@
+# SageMaker Code Editor Auto-Stop for Idle Instances
+
+The `autostop.py` Python script, coupled with the `on-start.sh` shell script, is designed to automatically shut down idle SageMaker Code Editor applications after a configurable time of inactivity. This solution is intended to help manage costs by ensuring that resources are not left running when not in use.
+
+## Installation for SageMaker Studio User Profiles
+
+### Prerequisites
+
+- AWS CLI configured with appropriate permissions
+- Access to the SageMaker Studio domain where the user profiles are located
+
+### Installation for all user profiles in a SageMaker Studio domain
+
+From a terminal appropriately configured with AWS CLI, run the following commands (replace fields as needed):
+
+```
+ASI_VERSION=0.3.0
+
+curl -LO https://github.com/aws-samples/sagemaker-studio-apps-lifecycle-config-examples/releases/download/v$ASI_VERSION/code-editor-start-$ASI_VERSION.tar.gz
+tar -xvzf code-editor-start-$ASI_VERSION.tar.gz
+
+cd auto_shut_down
+
+REGION=<aws_region>
+DOMAIN_ID=<domain_id>
+ACCOUNT_ID=<aws_account_id>
+LCC_NAME=code-editor-auto-shut-down
+LCC_CONTENT=`openssl base64 -A -in on-start.sh`
+
+aws sagemaker create-studio-lifecycle-config \
+    --studio-lifecycle-config-name $LCC_NAME \
+    --studio-lifecycle-config-content $LCC_CONTENT \
+    --studio-lifecycle-config-app-type CodeEditor \
+    --query 'StudioLifecycleConfigArn'
+
+aws sagemaker update-domain \
+    --region "$REGION" \
+    --domain-id "$DOMAIN_ID" \
+    --default-user-settings \
+    '{
+      "CodeEditorAppSettings": {
+        "DefaultResourceSpec": {
+          "LifecycleConfigArn": "arn:aws:sagemaker:'"$REGION"':'"$ACCOUNT_ID"':studio-lifecycle-config/'"$LCC_NAME"'",
+          "InstanceType": "ml.t3.medium"
+        },
+        "LifecycleConfigArns": [
+          "arn:aws:sagemaker:'"$REGION"':'"$ACCOUNT_ID"':studio-lifecycle-config/'"$LCC_NAME"'"
+        ]
+      }
+    }'
+
+```
+
+2. After successful domain update, navigate to Code Editor, and select the LCC when starting your Code Editor application.
+
+
+### Definition of idleness
+
+The current implementation of idleness (as of `v0.3.0`) includes the following criteria:
+
+1. There are no file changes made in the Code Editor application for a time period greater than `IDLE_TIME`. File changes include adding new files, deleting files, and/or updating files.
+
```

Comment:
We should add a note explaining that terminal activity is not detected at the moment.

Reply:
Draft for addition (at line 62):

"Please note that the `IDLE_TIME` calculation does not start at the moment the Code Editor application starts, but there is a latency time of at least 2 minutes for when the `crontab` starts. For more information, please see: https://cronhub.io/?utm_source=crontab&utm_medium=referral&utm_campaign=sideproject-campaign"

Reply:
Draft for README addition on line 62:

"* Note: As of `v0.3.0`, there is no detection of terminal activity to account for idleness."

Reply:
I would also add a statement explaining the impact of this. Something like: "This implies that, as an example, if a user executes a long-running script in the terminal and waits without performing any further action, and the execution takes more than the configured IDLE_TIME, the Code Editor application will be terminated."

Reply:
We'll add this in a separate PR.